recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java

```diff
@@ -37,22 +43,58 @@ public static boolean isPublic(Recipe recipe) {
     public List<RecipeDTO> getAllPublicRecipes() {
         return recipeMapper.map(recipeRepository.findAll().stream()
                 .filter(RecipeServiceImpl::isPublic)
-                .collect(Collectors.toList()));
+                .toList());
     }
 
     @Override
     public List<RecipeDTO> getAllPublicRecipesByName(String recipeName) {
         return  recipeMapper.map(recipeRepository.findAll().stream()
                 .filter(RecipeServiceImpl::isPublic)
                 .filter(r -> r.getName().equals(recipeName))
-                .collect(Collectors.toList()));
+                .toList());
     }
 
     @Override
     public List<RecipeDTO> getAllPublicRecipesByType(MealType mealType) {
         return recipeMapper.map(recipeRepository.findAll().stream()
                 .filter(RecipeServiceImpl::isPublic)
                 .filter(r -> r.getMealType() == mealType)
-                .collect(Collectors.toList()));
+                .toList());
+    }
+
+    @Override
+    @Transactional
+    public Recipe createRecipe(RecipeRequestDTO recipeRequestDTO, Long userId) {
+
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found"));
+        if (user.getRole() != Role.Admin )
+            throw new RuntimeException("Only admins can create recipes");
+
+        Recipe recipe = recipeMapper.toEntity(recipeRequestDTO);
+
+        recipe.setUser(user);
+
+        Recipe savedRecipe = recipeRepository.save(recipe);
+
+        List<RecipeIngredient> recipeIngredients = recipeRequestDTO.getIngredients().stream()
+                .map(ingredientDTO -> {
+                    RecipeIngredient recipeIngredient = new RecipeIngredient();
+                    recipeIngredient.setRecipe(savedRecipe);
+
+                    UserIngredient ingredient = userIngredientRepository.findById(ingredientDTO.getIngredientId())
+                            .orElseThrow(() -> new RuntimeException("Ingredient not found"));
```

Comment:
For handling not found values in database, is recommanded to be used new EntityNotFoundException("No ingredient found in the DB.") instead of RuntimeException